RFC/src/RFC-0345_AssetLifeCycle.md

```diff
@@ -0,0 +1,145 @@
+
+* The validator node committee for the asset must support the migration path. This entails that _every_ VN in the
+  committee has a `MigrateAsset` class from the `template_type.version` combination of the existing asset to the
+  `template_type.version` of the new asset.
+* The original asset issuer provides a valid `migrate_asset` instruction to the DAN.
+  * The asset issuer MUST provide any additional state that exists in the new template version and not the original.
+  * The original asset SHOULD be marked as `retired`. If so, the `superseded_by` field in the old asset will carry the
+    new asset id once the new asset has been confirmed. We recommend retiring the old asset because all the keys that
+    indicate ownership of tokens will be copied over; effectively re-using them; which can damage privacy.
+  * A policy is provided to determine the course of action to follow if any state from the old asset is illegal under
+    the new template rules (e.g. If a new rule requires `token.age` to be > 21; what happens to any tokens where this
+    condition fails?)
+
+
+As part of the migration,
+
+1. An entirely new asset is created with the full state of the old asset copied over into the new asset; supplemented
+   with any additional state required in the new template.
+2. A state validation run is performed; and any invalid state from the old asset is modified according to the migration
+  policy.
+3. Step 2 is repeated until a valid initial state is produced.
+4. If Step 2 has run `STATE_VALIDATION_LIMIT` times and the initial state is still not valid, the migration instruction
+   is rejected; the migrate_asset instruction will advise what should be done with the original asset in this case:
+   either allow the original asset to continue as before, or retire it.
+5. Once a valid initial state is produced, a new `create_asset` instruction is generated from the initial
+   state and the `migrate_asset` instruction. Typically the same VN committee will be used for the new asset, but this
+   needn’t be the case.
+
+Once a successful migration has completed, any instructions to the old asset can return a simple `asset_migrated`
+response with the new asset ID, which will allow clients and wallets to seamlessly update their records.
+
+### Retiring Assets
+
+Retiring an asset follows the same procedure as when as asset reaches its natural end-of-life: A final checkpoint is
+posted to the base layer and a grace period is given to allow DAN nodes and clients to take a snapshot of the final
+state if desired.
+
+### Resurrecting assets
+
+It’s unreasonable to expect VNs to hold onto large chunks of state for assets that are effectively dead (e.g. ticket
+stubs long after the event is over). For this reason, assets are allowed to expire after which VNs can forget about the
+state and use that storage for something else.
+
+However, it may be that interest in an asset resurfaces long after the asset expires (nostalgia being the multi-billion
```

Comment:
What about support for a proof of state, like for example RSA accumulators? 
If the interests are just in proving a certain state, this should be fine and it wont require a resurrecting.

Reply:
Yes, but that's a different use case altogether. If you wanted to prove that you went to the last Killers concert for example, you could potentially prove that with a Merklish proof and not need to resurrect the contract.

But if you wanted to sell that ticket stub (because it has nonzero value), the contract would need to be resurrected because transfers imply a state change.